app/Http/Controllers/Admin/StatisticsController.php

```diff
@@ -43,7 +43,11 @@ public function index(Request $request)
 
         $tokens = [];
         foreach ($nodes as $node) {
-            $tokens[$node->id] = $this->keyProviderService->handle($node->servers->get(0), $request->user());
+            $server = Server::where('node_id', $node->id)->first();
+            if ($server == null)
+                continue;
+
+            $tokens[$node->id] = $this->keyProviderService->handle($server, $request->user());
         }
 
         Javascript::put([
```

Comment:
There should be an `InjectsJavascript` trait, or similarly named one that will allow you to make this call unit testable.